### DISCUSSION TITLE
Can you add these 4 overlay buttons please, whenever you get time?

### DISCUSSION THREAD

**Discussion by Aryan-Chauhan** [anddea/revanced-patches#564] [General]
• #47 - Captions Menu

• #49 - Quality Menu

• #50 - Android's Native Share Menu

• #51 - Save To Playlist Menu

All 4 are important for me, so it's okay if it takes 1 year or more. Just add them, whenever you get time. Also, it is requested to keep the issues open.

**Comment by anddea** [comment]
Why can't you use built-in options? There's captions button, save ot playlist button, and you can open quality menu by tapping on timestamp info quality. 

What issue?

**Comment by Aryan-Chauhan** [comment]
Reason for Captions Overlay Button:

![Screenshot_2024-06-01-07-27-24-14_fe557b5c831afa8eecca413bd340ebe7.jpg](https://github.com/anddea/revanced-patches/assets/91815107/5493d44d-e68a-4259-a727-ac1ac7e7990d)

Currently, there is one button on the left & two buttons on the right. This looks asymmetrical. This is why captions button on the overlay buttons bar will look better.

**Comment by Aryan-Chauhan** [comment]
Reason for Quality Overlay Button:

Timestamp is a text which is horizontally long, but vertically short. That's why it's hard to tap on it for some like me with big fingers. On the other hand, I can easily use overlay buttons, because they are sufficiently long vertically & horizontally.

That's why an overlay button that opens up Quality Menu will be helpful for me.

**Comment by Aryan-Chauhan** [comment]
Reason for Android's Native Share Button:

It will bypass the in-built YT share menu, which has very less options & provide instant convenience. This is the main motivation.

If implemented, people can also keep the `Copy Link overlay button` turned off, because Android's Native Share Menu already has an option to Copy Link. So, two tasks can be done with single button. This is secondary motivation.

**Comment by Aryan-Chauhan** [comment]
Reason for Save to Playlist Button:

In player action buttons, there is inconsistent padding/spacing between them. This does not look good & always attracts attention. That's why I keep all player action buttons hidden. But this keeps me deprived from the Save button. That's why Save to Playlist overlay button will be helpful for me. It will solve the padding/spacing issue.
![Screenshot_2024-06-01-07-58-32-01_fe557b5c831afa8eecca413bd340ebe7.jpg](https://github.com/anddea/revanced-patches/assets/91815107/0f842865-ab49-4fb4-9254-8529dfe69161)

**Comment by Aryan-Chauhan** [comment]
**Common reason for all 4 overlay buttons:**

It will solve the inconsistent spacing/padding issue that exists between player action buttons when you hide some of them.

https://github.com/ReVanced/revanced-patches/issues/62

https://github.com/ReVanced/revanced-patches/issues/393

https://github.com/ReVanced/revanced-patches/issues/1653 **[OPEN ISSUE]**

https://github.com/inotia00/ReVanced_Extended/issues/17

https://github.com/inotia00/ReVanced_Extended/issues/89

https://github.com/inotia00/ReVanced_Extended/issues/451

https://github.com/inotia00/ReVanced_Extended/issues/674

https://github.com/inotia00/ReVanced_Extended/issues/1754

Because of above issues, I have to use my app like this:
![Screenshot_2024-03-17-08-43-18-88_f22faf49515899f9781ec174a4d85388.jpg](https://github.com/anddea/revanced-patches/assets/91815107/2ed84044-2f18-4e30-aba0-64c1f83b734d)

If the inconsistent padding/spacing issue is solved, then at least 2 of the suggested button aren't required.
